wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/Generic/Fees.hs

```diff
@@ -88,23 +88,41 @@ senderPaysFee :: (Monad m, CoinSelDom (Dom utxo))
                    ([CoinSelResult (Dom utxo)], SelectedUtxo (Dom utxo))
 senderPaysFee pickUtxo totalFee css = do
     let (css', remainingFee) = feeFromChange totalFee css
-    (css', ) <$> coverRemainingFee pickUtxo remainingFee
+    (additionalUtxo, additionalChange) <- coverRemainingFee pickUtxo remainingFee
+    let css'' = returnAdditioncalChange css' additionalChange
+    return (css'', additionalUtxo)
```

Comment:
To be honest, I'd rather add that logic to `runCoinSelT` from `Cardano.Wallet.Kernel.CoinSelection.FromGeneric`.

When we construct the final result:

```hs
        let css  = map (coinSelRemoveDust (csoDustThreshold opts)) cssWithDust
            inps = concatMap selectedEntries
                     (additionalUtxo : map coinSelInputs css)
            outs = map coinSelOutput css
        let allInps = case inps of
                        []   -> error "runCoinSelT: empty list of inputs"
                        i:is -> i :| is
            originalOuts = case outs of
                               []   -> error "runCoinSelT: empty list of outputs"
                               o:os -> o :| os
        return . Right $ CoinSelFinalResult allInps
                                            originalOuts
                                            (concatMap coinSelChange css)
```

This would avoid fiddling with `CoinSelResult` here and "artificially" add the change. To my understanding, we've separated additional inputs yielded by the fee policy for a reason; We have conflated them with the `CoinSelResult` inputs already but we didn't. Instead, we reconcialte the `CoinSelFinalResult` later and therefore, I see this as a similar case.

Reply:
Should I create an additional change just for these additional UTXO's instead of just adding it to the existing one? In this case should I also check if this new change is dust? Not sure :thinking:

Reply:
Cf @edsko inputs I believe. Let's distribute the extra change across all changes.